Assets/ScriptsWithDependencies/RegistrationManager.cs

```diff
@@ -37,17 +42,22 @@ public class RegistrationManager : MonoBehaviour
     void Start()
     {
         emailRegex = new Regex("^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$");
-        userRegex = new Regex("^.{5,20}$");
-        passwordRegex = new Regex("^.{5,20}$");
+        userRegex = new Regex("^.{3,20}$");
+        passwordRegex = new Regex("^.{6,20}$");

```

Comment:
Passwort sollte eigentlich auf minimum 8 Zeichen sein. Heutzutage sind die Rechner 8 Core 2.8Ghz, d.h. 1.7*10^-6.

Ein 8 Charakter, full lowercase passwort würde (1.7*10^-6 * 26^8) dauern, also c.a. 2 Tage, aber bei 6 Charaktern dauert es durch bruteforce wenige Sekunden. 

Wenn wir im Regex noch hinzufügen, dass ein Spezialcharakter dabei ist und eine Zahl, dann haben wir c.a. 80 Charaktere, d.h. (1.7*10^-6 * 80^8), also c.a. 45 Jahre an einem nicht Superrechner.

Wir könnten uns auch überlegen das Passwort auf minimum 10 Charaktere zu setzen mit Spezialcharakter etc., dann würde das Passwort in (1.7*10^-6 * 80^10), also c.a. in 289217 Jahren gecracked.

Also ja, bitte mindestens Regex für Passwort auf 8 Charaktere wechseln.
```suggestion
        passwordRegex = new Regex("^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,20}$");
```
Minimum 8 Charaktere, 1 Grossbuchstabe, 1 Kleinbuchstabe, 1 Symbol.

Reply:
Manche Menschen wollen das Password "123456" verwenden. Das ist mit deinem Regex unmöglich.